### ISSUE TITLE
VideoComponent object causes immediate crash on iGPU.

### ISSUE THREAD

**Issue by Jacksaur** [mmatyas/pegasus-frontend#908]
### The issue

Every time I run pegasus, the loading screen appears, the bar instantly reaches around a quarter of progress, and the software instantly closes again. Initially the log only got up to importing GOG games, so I disabled it and the "No games found" screen would load up fine.
I then added some games, and the problem started again. This time the log ends with "Requested theme `:/themes/pegasus-theme-grid` not found, falling back to default" every time. There was no themes folder entirely, so I downloaded the grid theme and placed it in that location, but the same still occurs.
Interestingly it did start once, and I could navigate the games list and everything fine. Once I closed it, the problem immediately resumed again.

### System info

Pop! OS Linux with KDE Plasma desktop environment.

### Pegasus version

Can't even view it, but I got the latest download available on the site.

### Pegasus log
2021-10-14T15:48:37 [i] Pegasus alpha15-55-gb2834d3c (2021-09-25)
2021-10-14T15:48:37 [i] Running on Pop!_OS 21.04 (x86_64, xcb)
2021-10-14T15:48:37 [i] Qt version 5.15.2
2021-10-14T15:48:37 [i] Program settings loaded (`/home/james/Tools/Pegasus/config/settings.txt`)
2021-10-14T15:48:37 [i] Found locale `ar`
2021-10-14T15:48:37 [i] Found locale `bs`
2021-10-14T15:48:37 [i] Found locale `de`
2021-10-14T15:48:37 [i] Found locale `en-GB`
2021-10-14T15:48:37 [i] Found locale `en`
2021-10-14T15:48:37 [i] Found locale `es`
2021-10-14T15:48:37 [i] Found locale `fr`
2021-10-14T15:48:37 [i] Found locale `hu`
2021-10-14T15:48:37 [i] Found locale `ja`
2021-10-14T15:48:37 [i] Found locale `ko`
2021-10-14T15:48:37 [i] Found locale `lt`
2021-10-14T15:48:37 [i] Found locale `nl`
2021-10-14T15:48:37 [i] Found locale `pt-BR`
2021-10-14T15:48:37 [i] Found locale `ru`
2021-10-14T15:48:37 [i] Found locale `zh-TW`
2021-10-14T15:48:37 [i] Found locale `zh`
2021-10-14T15:48:37 [i] Locale set to `en-GB`
2021-10-14T15:48:37 [i] Found theme `Pegasus Grid` at `:/themes/pegasus-theme-grid/`
2021-10-14T15:48:37 [i] Found theme `Pegasus Grid` at `/home/james/Tools/Pegasus/themes/pegasus-theme-grid/`
2021-10-14T15:48:37 [w] Requested theme `:/themes/pegasus-theme-grid` not found, falling back to default

edit:
I reenabled the GOG search after reading the thread about issues caused by it, just to see if it'd have an effect. Pegasus still closes almost immediately on startup same as usual, but the log is longer at least.

2021-10-14T22:09:19 [i] Pegasus alpha15-85-gfff1a5b2 (2021-10-04)
2021-10-14T22:09:19 [i] Running on Pop!_OS 21.04 (x86_64, xcb)
2021-10-14T22:09:19 [i] Qt version 5.15.2
2021-10-14T22:09:19 [i] Program settings loaded (`/home/james/Tools/Pegasus/config/settings.txt`)
2021-10-14T22:09:19 [i] Found locale `ar`
2021-10-14T22:09:19 [i] Found locale `bs`
2021-10-14T22:09:19 [i] Found locale `de`
2021-10-14T22:09:19 [i] Found locale `en-GB`
2021-10-14T22:09:19 [i] Found locale `en`
2021-10-14T22:09:19 [i] Found locale `es`
2021-10-14T22:09:19 [i] Found locale `fr`
2021-10-14T22:09:19 [i] Found locale `hu`
2021-10-14T22:09:19 [i] Found locale `ja`
2021-10-14T22:09:19 [i] Found locale `ko`
2021-10-14T22:09:19 [i] Found locale `lt`
2021-10-14T22:09:19 [i] Found locale `nl`
2021-10-14T22:09:19 [i] Found locale `pt-BR`
2021-10-14T22:09:19 [i] Found locale `ru`
2021-10-14T22:09:19 [i] Found locale `zh-TW`
2021-10-14T22:09:19 [i] Found locale `zh`
2021-10-14T22:09:19 [i] Locale set to `en-GB`
2021-10-14T22:09:19 [i] Found theme `Pegasus Grid` at `:/themes/pegasus-theme-grid/`
2021-10-14T22:09:19 [i] Found theme `Pegasus Grid` at `/home/james/Tools/Pegasus/themes/pegasus-theme-grid/`
2021-10-14T22:09:19 [w] Requested theme `:/themes/pegasus-theme-grid` not found, falling back to default
2021-10-14T22:09:19 [i] Theme set to `Pegasus Grid` (`:/themes/pegasus-theme-grid/`)
2021-10-14T22:09:19 [i] SDL version 2.0.16
2021-10-14T22:09:19 [i] Metafiles: Found `/home/james/Tools/Pegasus/Games/NES/metadata.pegasus.txt`
2021-10-14T22:09:19 [i] Metafiles: Finished searching in 2ms
2021-10-14T22:09:19 [i] Media: Finished searching in 1ms
2021-10-14T22:09:19 [i] GOG: 0 games found
2021-10-14T22:09:19 [i] GOG: Finished searching in 0ms
2021-10-14T22:09:19 [i] EmulationStation: No installation found
2021-10-14T22:09:19 [i] EmulationStation: Finished searching in 0ms
2021-10-14T22:09:19 [i] Skraper Assets: 0 assets found
2021-10-14T22:09:19 [i] Skraper Assets: Finished searching in 0ms
2021-10-14T22:09:19 [i] Favorites: Finished searching in 0ms
2021-10-14T22:09:19 [i] Playtime: Finished searching in 0ms
2021-10-14T22:09:19 [w] The collection 'GOG' has no valid games, ignored

**Comment by mmatyas** [comment]
Hm, I just tried it out on Pop 21.04 with version alpha15-85-gfff1a5b2 downloaded from the website, and it worked fine for me. Interestingly, you have both

```
2021-10-14T15:48:37 [i] Found theme Pegasus Grid at :/themes/pegasus-theme-grid/
```

and

```
2021-10-14T22:09:19 [w] Requested theme :/themes/pegasus-theme-grid not found, falling back to default
```

` :/themes/pegasus-theme-grid` is built into the executable, so it should be always available, but apparently doesn't work in your case. This makes Pegasus crash at the moment it tries to load the theme files.

Could you post your `/home/james/Tools/Pegasus/config/settings.txt`? If you delete it, does that help?

**Comment by Jacksaur** [comment]
Video seems to be the likely cause. Whilst I don't actually have any video media for my games whatsoever, GameOS does have an option for a video background at the top, and SkylineOS doesn't seem to show videos in its theme at all: So my guess is that the mere act of initializing an area for videos is what causes the crash?

The default theme, does that play videos in the background normally? I tried starting it a few times again, crashed on load first two times as usual, third time it opened and worked fine, but when I switched to the Steam tab, the background began to fade in and crashed it right there. Log gave the same "Could not create wrappped context for platform" error as before. If it does, then I guess that would certify that it's videos causing the crash somehow. Perhaps because of my iGPU.

I don't have any GOG games, or the client itself, installed at all. So I'm clueless as to why that's affecting anything. But I don't really intend to use GOG at all, I mainly wanted Pegasus as a replacement frontend so I could move away from Launchbox. So I'm content with just disabling the GOG provider at all times.

**Comment by barolo** [comment]
I didn't want to open new issue but I'm getting the wrapped-context error myself, I'm on Gentoo, with everything latest and AMD GPU
I don't think that it's affecting most of the themes though, they all run fine I think.
```
** (pegasus-fe:29950): CRITICAL **: 21:26:35.447: gst_gl_context_new_wrapped: assertion 'available_apis != GST_GL_API_ANY' failed
[w] Could not create wrappped context for platform: 2

```

**Comment by mmatyas** [comment]
Hi, are you using Wayland? Apparently it's related to some issues with GStreamer's Wayland port in Qt Multimedia.

**Comment by barolo** [comment]
Nope, It's under X11

**Comment by Jacksaur** [comment]
X11 on my end as well I'm afraid.

**Comment by Jacksaur** [comment]
After distro hopping to Kubuntu 21.10 and reinstalling under Flatpak, the program now works perfectly and videos even play without any problems. I'm guessing that Flatpak has fixed it, rather than the distro hop.
Only thing left is I'm having trouble finding where to restore my themes and config under the flatpak version. I've searched around the /var/lib/flatpak/app/ but can't find either location. Tried the .config/pegasus-frontend location mentioned on the site but no dice.
